package.json

```diff
@@ -24,7 +24,8 @@
     "cover": "nyc jasmine --config=spec/coverage.json",
     "e2e-tests": "jasmine --config=spec/e2e/jasmine.json",
     "java-unit-tests": "node test/run_java_unit_tests.js",
-    "eslint": "eslint bin spec test"
+    "eslint": "eslint bin spec test",
+    "clean": "gulp clean"
```

Comment:
Perhaps this may be achieved without adding gulp as a dependency (via `clean.js`)?

Reply:
Project search for `clean.js` yielded no results.

I have found `templates/cordova/lib/clean` however but this is for cleaning the project directory, not the test directory.

To be clear, the `clean` npm script is only for development on the `cordova-android` to reset gradle. It's not for anybody who uses `cordova-android` as part of their project via `cordova platform add`.

Maybe I should write a comment explaining as such? I believe this is the second time there was a misunderstanding.

Reply:
Also worth mentioning that gradle doesn't need to be "cleaned" for cordova apps because to upgrade the `cordova-android` platform, you need to remove the platform (and thus destroying the gradle installation with it) and then re-adding the platform, which will install whatever gradle version as defined in `cordova-android`.

Reply:
What I meant is that the contents of [gulpfile.js](https://github.com/apache/cordova-android/pull/792/files#diff-b9e12334e9eafd8341a6107dd98510c9) could be moved to separate script - for example `bin/clean.js` which could be run via npm.

`package.json`:

```
{
  "scripts": {
    "clean": "node bin/clean.js"
  }
}
```

Reply:
Ahh gotcha. I can move the `gulpfile` to `bin/`. Although gulp is a task system, so you would run the file with `gulp`, and not node directly and I would favour keeping the filename `gulpfile.js` because of this. Reason for adding this tool is to have easier script to clean the test gradle stuff, without relying on `git` commands. It's a dependency that some other cordova repos already use so it should be somewhat familiar.

Reason this is being added as part of this PR is because it does require gradle to be removed for the test directory to use the new version of gradle. This would only apply to other developers who already have this repository cloned and they have ran `npm test` before.

Reply:
I am also not in favor of adding the `gulp` dependency to perform a simple task that could have been achieved with just node.

I will let someone else though review this.

Reply:
> I am also not in favor of adding the `gulp` dependency to perform a simple task that could have been achieved with just node.
> 
> I will let someone else though review this.

Fair enough. My thought process was to use something existing for familiarity. But it is true that this isn't solving a complex problem so `gulp` isn't really as necessary compared to `cordova-docs` (where `gulp` is used).

I can remove `gulp` dependency and write a simple node script to remove that files that needs to be removed if you want gradle to reinstall for `npm test` and I can place this node script in the `bin/` directory which I believe will solve both critiques.

Does this sound right?

Reply:
Sounds good to me!

Reply:
I've made the changes as discussed. Resolving this conversation.

Reply:
> I've made the changes as discussed.

18a9e7d72615e748be01dde3a65a4a7c3ee14383 - rare case where removing a dependency results in *less* script code (sorry if this is too far off-topic)